src/main/java/org/jfrog/hudson/util/BuildRetentionFactory.java

```diff
@@ -24,7 +25,14 @@
     public static BuildRetention createBuildRetention(Run build, boolean discardOldArtifacts) {
         BuildRetention buildRetention = new BuildRetention(discardOldArtifacts);
         LogRotator rotator = null;
-        BuildDiscarder buildDiscarder = build.getParent().getBuildDiscarder();
+        BuildDiscarder buildDiscarder;
```

Comment:
Question - doesn't this functionality fail if the matrix job plugin is not installed?
If it does, you can see what we did to support the *jenkins-multijob-plugin*:
```
    public boolean isApplicable(AbstractProject<?, ?> item) {
        this.item = item;
        Class<?> itemClass = item.getClass();
        return itemClass.isAssignableFrom(FreeStyleProject.class) || itemClass.isAssignableFrom(MatrixProject.class) ||
                (Jenkins.getInstance().getPlugin(PluginsUtils.MULTIJOB_PLUGIN_ID) != null &&
                        itemClass.isAssignableFrom(MultiJobProject.class)) ||
                PluginsUtils.PROMOTION_BUILD_PLUGIN_CLASS.equals(itemClass.getSimpleName());
    }
```